modules/Bio/EnsEMBL/Hive/RunnableDB/FastaFactory.pm

```diff
@@ -130,26 +131,35 @@ sub write_output {
 
     my $input_seqio         = $self->param('input_seqio');
     my $max_chunk_length    = $self->param('max_chunk_length');
+    my $output_dir          = $self->param('output_dir');
     my $output_prefix       = $self->param('output_prefix');
     my $output_suffix       = $self->param('output_suffix');
 
     my $chunk_number = 1;   # counts the chunks
     my $chunk_length = 0;   # total length of the current chunk
     my $chunk_size   = 0;   # number of sequences in the current chunk
-    my $chunk_name   = $output_prefix.$chunk_number.$output_suffix;
+    
+    my $current_output_directory;    
     if ($self->param('hash_directories')) {
         my $dir_tree = dir_revhash($chunk_number);
-        if ($dir_tree ne '') {
-            mkpath($dir_tree);
-            $chunk_name = $dir_tree.'/'.$chunk_name;
+        
+        if ($dir_tree eq '') {
+	  $current_output_directory = $output_dir;
+        } else {
+	  $current_output_directory = $output_dir . '/' . $dir_tree;
         }
+    } else {      
+      $current_output_directory = $output_dir;
```

Comment:
The block can probably look cleaner if `$current_output_directory` is initialized to `$output_dir` and modified when needed